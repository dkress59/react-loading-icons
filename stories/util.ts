import { ArgTypes } from '@storybook/react'
import { SVGProps } from 'react'

export type IconProps = JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
export const IconFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: 'transparent',
	strokeOpacity: 1,
	width: 'auto',
}
export const IconStroke: IconProps = {
	fill: 'transparent',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}
export const IconStrokeFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}

export const IconFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: 'transparent',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: 'transparent',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
