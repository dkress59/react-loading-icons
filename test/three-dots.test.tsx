import { render, screen } from '@testing-library/react'
import React from 'react'

import ThreeDots from '../src/components/three-dots'
import { testProps } from './util'

describe('<ThreeDots /> component', () => {
	it('renders and matches snapshot', () => {
		render(<ThreeDots data-testid="icon" />)
		const svg = screen.getByTestId('icon')
		expect(svg).not.toBeNull()
		expect(svg).toMatchSnapshot()
		expect(svg.getAttributeNames()).toContain('viewBox')
		expect(svg.getAttribute('class')).toBe('icon-loading')
	})
	it('inherits props', () => {
		render(<ThreeDots {...testProps} />)
		const svg = screen.getByTestId('icon')
		expect(svg.getAttribute('height')).toBe('2em')
		expect(svg.getAttribute('width')).toBe('auto')
		expect(svg.getAttribute('class')).toBe('icon-loading test')
	})
})
