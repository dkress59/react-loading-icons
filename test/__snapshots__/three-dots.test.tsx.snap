// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThreeDots /> component renders and matches snapshot 1`] = `
<svg
  class="icon-loading"
  data-testid="icon"
  height="30"
  viewBox="0 0 120 30"
  width="120"
>
  <circle
    cx="15"
    cy="15"
    fill="#fff"
    r="15"
  >
    <animate
      attributeName="r"
      begin="0s"
      calcMode="linear"
      dur="0.8s"
      from="15"
      repeatCount="indefinite"
      to="15"
      values="15;9;15"
    />
    <animate
      attributeName="fill-opacity"
      begin="0s"
      calcMode="linear"
      dur="0.8s"
      from="1"
      repeatCount="indefinite"
      to="1"
      values="1;.5;1"
    />
  </circle>
  <circle
    cx="60"
    cy="15"
    fill="#fff"
    r="9"
  >
    <animate
      attributeName="r"
      begin="0s"
      calcMode="linear"
      dur="0.8s"
      from="9"
      repeatCount="indefinite"
      to="9"
      values="9;15;9"
    />
    <animate
      attributeName="fill-opacity"
      begin="0s"
      calcMode="linear"
      dur="0.8s"
      from="0.5"
      repeatCount="indefinite"
      to="0.5"
      values=".5;1;.5"
    />
  </circle>
  <circle
    cx="105"
    cy="15"
    fill="#fff"
    r="15"
  >
    <animate
      attributeName="r"
      begin="0s"
      calcMode="linear"
      dur="0.8s"
      from="15"
      repeatCount="indefinite"
      to="15"
      values="15;9;15"
    />
    <animate
      attributeName="fill-opacity"
      begin="0s"
      calcMode="linear"
      dur="0.8s"
      from="1"
      repeatCount="indefinite"
      to="1"
      values="1;.5;1"
    />
  </circle>
</svg>
`;
